pkg/resources/kafka/kafka.go

```diff
@@ -1217,6 +1235,44 @@ func (r *Reconciler) getK8sAssignedNodeport(log logr.Logger, eListenerName strin
 	return nodePort, nil
 }
 
+func (r *Reconciler) getK8sAssignedNodeAddress(brokerId int32, nodeAddressType string) (string, error) {
+	podList := &corev1.PodList{}
+	if err := r.Client.List(context.TODO(), podList,
+		client.InNamespace(r.KafkaCluster.Namespace),
+		client.MatchingLabels(apiutil.MergeLabels(apiutil.LabelsForKafka(r.KafkaCluster.Name), map[string]string{v1beta1.BrokerIdLabelKey: strconv.Itoa(int(brokerId))})),
+	); err != nil {
+		return "", err
+	}
+
+	switch {
+	case len(podList.Items) == 0:
+		return "", fmt.Errorf("could not find pod with matching brokerId: %d in namespace '%s'", brokerId, r.KafkaCluster.Namespace)
+	case len(podList.Items) > 1:
+		return "", fmt.Errorf("multiple pods found with brokerId: %d in namespace '%s'", brokerId, r.KafkaCluster.Namespace)
+	default:
+		return r.getK8sNodeIP(podList.Items[0].Spec.NodeName, nodeAddressType)
+	}
+}
+
+func (r *Reconciler) getK8sNodeIP(nodeName string, nodeAddressType string) (string, error) {
+	node := &corev1.Node{}
+	clientNamespacedName := types.NamespacedName{Name: nodeName, Namespace: r.KafkaCluster.Namespace}
+	if err := r.Client.Get(context.TODO(), clientNamespacedName, node); err != nil {
+		return "", err
+	}
+
+	addressMap := make(map[string]string)
+	for _, address := range node.Status.Addresses {
+		addressMap[string(address.Type)] = address.Address
+	}
+
+	if val, ok := addressMap[nodeAddressType]; ok && val != "" {
+		return val, nil
+	}
+
+	return "", fmt.Errorf("Could not find the node (%s)'s (%s) address", nodeName, nodeAddressType)
```

Comment:
```suggestion
	return "", fmt.Errorf("could not find the node (%s)'s (%s) address", nodeName, nodeAddressType)
```

See https://github.com/golang/go/wiki/CodeReviewComments#error-strings